### DISCUSSION TITLE
How do I dynamically create and delete a Topic in C++ or Node?

### DISCUSSION THREAD

**Discussion by JanYork** [apache/rocketmq#8037] [Q&A] [answered]
Can other languages have the ability to operate RocketMQ Admin? For example: adding or deleting Topic in C++ program.

It can be implemented in Java. Netty from rocket.remoting provides the ability to operate admin. You can use the tools in the rocket.tool package to implement management operations.

But there is no support for Netty in other languages. Netty is a Java library. Is there a better request method? Is it possible to use Grpc to implement the operational capabilities of rocketmq admin?

**Comment by biningo** [comment]
Admin has no sdk for other languages. But I think there are two ways to solve this problem.
1. You can wrap rocketmq-tools to implement the http proxy，and then use other language to request this http api. (it is easy)
2. Implement the RocketMQ protocol in other languages.

**Reply by biningo** [reply]
@JanYork But I don't know if [RIP-17](https://github.com/apache/rocketmq/wiki/RIP-17-RocketMQ-HTTP-Proxy-Support) will add an http api about admin.

**Comment by cserwen** [comment, marked as answer]
Currently, grpc maybe not support to operate metadata, you can implement the protocol through reference[ rocketmq-client-cpp ](https://github.com/apache/rocketmq-client-cpp) 

We can also implement RocketMQ admin for the GRPC protocol, but it doesn't seem to be achievable quickly
